code/datums/logging/_log_entry.dm

```diff
@@ -0,0 +1,177 @@
+/datum/log_entry
+	/*
+	 * The version of this log entry, based on semver: https://semver.org/
+	 *
+	 * This is used to guarantee to automated tooling that the data it is parsing
+	 * is what it is expecting.
+	 *
+	 * For backwards incompatible changes, such as REMOVING fields, or CHANGING their meaning,
+	 * update the MAJOR version number. For instance, 1.2.3 -> 2.0.0.
+	 *
+	 * For backwards compatible additions, such as ADDING fields,
+	 * update the MINOR version number. For instance, 1.2.3 -> 1.3.0.
+	 *
+	 * Otherwise, if you are changing the datum AT ALL, such as to refactor its code or fix an edge case,
+	 * update the PATCH version number. For instance, 1.2.3 -> 1.2.4.
+	 * This is so that in the event your code breaks the logging, automated tooling can isolate it.ted for.
+	 */
+	VAR_PRIVATE/version = "0.0.0"
+	/// The unix timestamp this log entry was created.
+	VAR_PRIVATE/unix_timestamp = 0
+	/// The world.time this log entry was created.
+	VAR_PRIVATE/world_timestamp = 0
+	/// The round this log entry was created on.
+	VAR_PRIVATE/round_id = 0
+	/// The server this log entry was created on.
+	VAR_PRIVATE/server_name
+
+	// required settings
+
+	/// The category this log entry falls under.
+	var/category
+	var/can_inspect = FALSE
+	/// The message this log entry contains.
+	var/message
+	/// Whether this log entry is public or not.
+	var/private = FALSE
+	/// The location this log entry was created at.
+	var/list/location
+	/// A lazy list of tags that can be used to filter this log entry.
+	var/list/tags
+
+	// optional settings
+
+	/// The name of the datum that created this log entry.
+	var/source_name
+	/// The ckey, if applicable, of the user that created this log entry.
+	var/source_ckey
+	/// The reference to the source of this log entry; for the default inspection
+	var/datum/weakref/source_weakref
+	/// The name of the target datum, if applicable, of this log entry.
+	var/target_name
+	/// The ckey, if applicable, of the target datum, if applicable, of this log entry.
+	var/target_ckey
+
+	/// extended information about this log entry; associative list
+	var/list/extended_data
+
+/datum/log_entry/New(message)
+	..()
+	SHOULD_CALL_PARENT(TRUE)
+	unix_timestamp = world.realtime
+	world_timestamp = world.time
+	round_id = GLOB.round_id
+	server_name = CONFIG_GET(string/cross_comms_name) || "Unknown"
```

Comment:
This should use the server IP and port as a fallback, assuming that isn't sensitive now for any reason (I know for sure port isn't, but MSO might have some really cool setup where we're not exposing the real IP or something, I'll check when this is undrafted).

Reply:
``CONFIG_GET(string/server)`` gets the URL + port, which in our case hides the real IP.

Reply:
On an unconfigured server, it will be null.
I haven't really looked at the implementation but it appears a bit redundant. if this saves the IP/URL/Servername/whatever in every single log entry. Those files already get bloated pretty bloated for raw text.

Reply:
It's been a while since we discussed, but one of the things me and oranges wanted was for logs to be fairly contextless. Compression should handle that pretty well, but yeah I definitely understand the waste perspective. I guess theoretically could be a config for anyone concerned about file space but I'm not sure it's that important and it also presents the ability for fields to be removed without version number going up.

Reply:
I see your point. Wasted space isn't really a concern if that's a priority.